### PR TITLE
MVKCmdClearAttachments: Account for the renderArea.

### PR DESCRIPTION
**Pull request by cdavis5e**
When calculating the vertices, we need to use the render area's
extent--but only if the implementation supports constraining the render
area using `renderTargetWidth` and `renderTargetHeight`. Otherwise, the
quad will be stretched and/or squashed because of the render area
constraint.

### CODE REVIEW

**Review by billhollings — APPROVED** [review]
👍🏻

File: MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm [hunk 1]
```diff
@@ -1130,6 +1130,15 @@ static inline MTLSize mvkClampMTLSize(MTLSize size, MTLOrigin origin, MTLSize ma
 	simd::float4 clearColors[kMVKClearAttachmentCount];
 
 	VkExtent2D fbExtent = cmdEncoder->_framebuffer->getExtent2D();
+	// I need to know if the 'renderTargetWidth' and 'renderTargetHeight' properties
+	// actually do something, but [MTLRenderPassDescriptor instancesRespondToSelector: @selector(renderTargetWidth)]
+	// returns NO even on systems that do support it. So we have to check an actual instance.
+	MTLRenderPassDescriptor* tempRPDesc = [MTLRenderPassDescriptor new];	// temp retain
+	if ([tempRPDesc respondsToSelector: @selector(renderTargetWidth)]) {
```
Comment (billhollings) [inline MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm:1137]: This is causing build warnings on tvOS. Need to wrap this section in `#if MVK_MACOS_OR_IOS`

**Review by billhollings — COMMENTED** [review]
@cdavis5e Build warnings on tvOS.

File: MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm (hunk 1 above)
Comment (cdavis5e) [reply MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm:1137]: #1063

**Pull request merged**